main.c

```diff
@@ -29,6 +33,13 @@ void fifo_handler(struct page_table *pt, int page ) {
     int frame;
     bool evict = true;
 
+    // Case: page does not have write permissions
+    if(pt->page_bits[page] == PROT_READ) {
+	page_table_set_entry(pt, page, pt->page_mapping[page], PROT_READ|PROT_WRITE);
+	num_faults++;
```

Comment:
If the permission bits are already set to read, the page fault must have been the result of an attempted write